main.tf

```diff
@@ -1,52 +1,48 @@
 
 
 resource "aws_vpc" "main" {
-  cidr_block = "10.0.0.0/16"
+  cidr_block = local.vpc_cidr
 
   tags = {
     "Name" = "custom-VPC"
   }
 }
 
-# Public subnets 
+locals {
+  vpc_cidr = "10.0.0.0/16"
+  public_cidr        = ["10.0.0.0/24", "10.0.1.0/24"]
+  private_cidr        = ["10.0.2.0/24", "10.0.3.0/24"]
 
-resource "aws_subnet" "public1" {
-  vpc_id            = aws_vpc.main.id
-  cidr_block        = "10.0.0.0/24"
-  availability_zone = "ap-south-1a"
-  tags = {
-    "Name" = "public-1"
-  }
+  availability_zones = ["ap-south-1a", "ap-south-1b"]
 }
 
-resource "aws_subnet" "public2" {
+# Public subnets 
+
+resource "aws_subnet" "public" {
+
+  count = 2
```

Comment:
could you replace the number 2 with `length(local.public_cidr)` for resources related to the public subnet and use 
`length(local.private_cidr)` for resources related to the private subnet as it makes the code more neat. 
i.e `count = length(local.public_cidr)` or `count = length(local.private_cidr)`

Reply:
@geoffrey1330 I have updated the main.tf file as suggested by you. now you can again review my PR . 
Thanks :)